File: specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/examples-java/ServerCommunicationLinkCreateOrUpdate.java
import com.azure.core.util.Context;
import com.azure.resourcemanager.sql.fluent.models.ServerCommunicationLinkInner;

/** Samples for ServerCommunicationLinks CreateOrUpdate. */
public final class Main {
    /*
     * x-ms-original-file: specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/examples/ServerCommunicationLinkCreateOrUpdate.json
     */
    /**
     * Sample code: Create a server communication link.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void createAServerCommunicationLink(com.azure.resourcemanager.AzureResourceManager azure) {
        azure
            .sqlServers()
            .manager()
            .serviceClient()
            .getServerCommunicationLinks()
            .createOrUpdate(
                "sqlcrudtest-7398",
                "sqlcrudtest-4645",
                "link1",
                new ServerCommunicationLinkInner().withPartnerServer("sqldcrudtest-test"),
                Context.NONE);
    }
}

File: specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/examples-python/ServerCommunicationLinkCreateOrUpdate.py
from azure.identity import DefaultAzureCredential
from azure.mgmt.sql import SqlManagementClient

"""
# PREREQUISITES
    pip install azure-identity
    pip install azure-mgmt-sql
# USAGE
    python server_communication_link_create_or_update.py

    Before run the sample, please set the values of the client ID, tenant ID and client secret
    of the AAD application as environment variables: AZURE_CLIENT_ID, AZURE_TENANT_ID,
    AZURE_CLIENT_SECRET. For more info about how to get the value, please see:
    https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal
"""


def main():
    client = SqlManagementClient(
        credential=DefaultAzureCredential(),
        subscription_id="00000000-1111-2222-3333-444444444444",
    )

    response = client.server_communication_links.begin_create_or_update(
        resource_group_name="sqlcrudtest-7398",
        server_name="sqlcrudtest-4645",
        communication_link_name="link1",
        parameters={"properties": {"partnerServer": "sqldcrudtest-test"}},
    ).result()
    print(response)


# x-ms-original-file: specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/examples/ServerCommunicationLinkCreateOrUpdate.json
if __name__ == "__main__":
    main()
